From e32ff2c2abc22532a58c9b687d922500d23fd709 Mon Sep 17 00:00:00 2001
From: Henrik Gramner <henrik at gramner.com>
Date: Thu, 16 Jan 2014 23:37:11 +0100
Subject: [PATCH] Use a larger hash table size

Increases compilation speed of files with a large number of macros by
around 20x.
---
 modules/preprocs/nasm/nasm-pp.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules/preprocs/nasm/nasm-pp.c b/modules/preprocs/nasm/nasm-pp.c
index 06eaa9a..27a8cc6 100644
--- a/modules/preprocs/nasm/nasm-pp.c
+++ b/modules/preprocs/nasm/nasm-pp.c
@@ -366,7 +366,7 @@ static ListGen *list;
  * FIXME: We should *really* be able to configure this at run time,
  * or even have the hash table automatically expanding when necessary.
  */
-#define NHASH 31
+#define NHASH 4096
 
 /*
  * The current set of multi-line macros we have defined.
-- 
2.0.3

